mrc-ide/packit
# Mrc 5604 single status endpoint

This adds the endpoint for the single status. Main changes include:
* get endpoint `status/{taskId}`
* adding new field queue position to run info table
* `status` in sql file has an extra `NOT NULL` after discussion in #110 
* `TaskStatus` data class, this is the orderly.runner response to status request

How to test:
1. `./scripts/dev-start --super-user` (with basic login enabled)
2. POST super user email and password `{"email": ..., "password": ...}` to `http://localhost:8080/auth/login/basic` to get the token (need this to make any requests to runner controller because it requires the `packet.run` permission), send the remaining requests with this bearer token
3. GET `http://localhost:8080/runner/git/branches` with token to get the `commitHash` (you'll need this later)
4. POST
```json
{
  "name": "parameters",
  "branch": "master",
  "hash": <YOUR-COMMIT-HASH>,
  "parameters": {
    "a": 6,
    "c": 28
  }
}
```
to `http://localhost:8080/runner/run` you will receive a `taskId`
5. GET `http://localhost:8080/runner/status/<YOUR-TASK-ID>` this should say completed status by this point and give the appropriate logs and details

OR you can wait for the FE so it will be easier to interact with it
